mm2src/coins_activation/src/prelude.rs

```diff
@@ -85,13 +98,35 @@ pub fn coin_conf_with_protocol<T: TryFromCoinProtocol>(
             (platform_conf, nft_protocol)
         },
         Err(_) => {
-            let conf = coin_conf(ctx, coin);
-            let coin_protocol: CoinProtocol = json::from_value(conf["protocol"].clone()).map_to_mm(|err| {
-                CoinConfWithProtocolError::CoinProtocolParseError {
-                    ticker: coin.into(),
-                    err,
-                }
-            })?;
+            let mut conf = coin_conf(ctx, coin);
+            let coin_protocol: CoinProtocol = match json::from_value(conf["protocol"].clone()) {
+                Ok(protocol_from_config) => {
+                    if let Some(protocol_from_request) = protocol_from_request {
+                        if protocol_from_request != protocol_from_config {
+                            return MmError::err(CoinConfWithProtocolError::ProtocolMissMatch {
+                                ticker: coin.into(),
+                                from_config: protocol_from_config,
+                                from_request: protocol_from_request,
+                            });
+                        }
+                    }
+                    protocol_from_config
+                },
+                Err(err) => match protocol_from_request {
```

Comment:
Can we work at all if "protocol" is missing in the 'coins' file?
I can see all coins there have "protocol" defined.

Reply:
No, we can't enable a coin from config without a defined protocol.